gpytorch/lazy/toeplitz_lazy_variable.py

```diff
@@ -191,113 +51,78 @@ def repeat(self, *sizes):
-                                            It is ambiguous which diagonal elements to choose')
-                    else:
-                        diag_new = None
-
-                return ToeplitzLazyVariable(self.c, J_left_new, C_left_new,
-                                            self.J_right, self.C_right, diag_new)
-            else:
-                # Batch modej
-                if self.c.ndimension() == 2:
-                    return ToeplitzLazyVariable(self.c[i], None, None, None, None, self.added_diag)
-                else:
-                    raise RuntimeError('Slicing an uninterpolated Toeplitz matrix to be non-square is probably \
-                                        unintended. If that was the intent, use evaluate() and slice the full matrix.')
+            return torch.Size((self.column.size(-1), self.column.size(-1)))
+
+    def _transpose_nonbatch(self):
+        return ToeplitzLazyVariable(self.column)
+
+    def __getitem__(self, index):
+        index = list(index) if isinstance(index, tuple) else [index]
+        ndimension = self.ndimension()
+        index += [slice(None, None, None)] * (ndimension - len(index))
+        column = self.column
+
+        squeeze_left = False
+        squeeze_right = False
+        if isinstance(index[-2], int):
+            index[-2] = slice(index[-2], index[-2] + 1, None)
+            squeeze_left = True
+        if isinstance(index[-1], int):
+            index[-1] = slice(index[-1], index[-1] + 1, None)
+            squeeze_right = True
+
+        # Handle batch dimensions
+        isbatch = ndimension >= 3
+        if isbatch:
+            batch_index = tuple(index[:-2])
+            column = self.column[batch_index]
+
+        ndimension = column.ndimension() + 1
+        print(ndimension)
+
+        # Handle index
+        left_index = index[-2]
+        right_index = index[-1]
+
+        batch_sizes = list(column.size()[:-1])
+        row_iter = column.data.new(column.size(-1)).long()
+        torch.arange(0, self.column.size(-1), out=row_iter)
+
+        left_interp_indices = row_iter[left_index].unsqueeze(-1)
+        right_interp_indices = row_iter[right_index].unsqueeze(-1)
+
+        left_interp_len = len(left_interp_indices)
+        right_interp_len = len(right_interp_indices)
+        for i in xrange(ndimension - 2):
+            left_interp_indices.unsqueeze_(0)
+            right_interp_indices.unsqueeze_(0)
+
+        left_interp_indices = left_interp_indices.expand(*(batch_sizes + [left_interp_len, 1]))
+        left_interp_values = left_interp_indices.new(left_interp_indices.size()).fill_(1).float()
+        right_interp_indices = right_interp_indices.expand(*(batch_sizes + [right_interp_len, 1]))
+        right_interp_values = right_interp_indices.new(right_interp_indices.size()).fill_(1).float()
+
+        res = InterpolatedLazyVariable(ToeplitzLazyVariable(column), Variable(left_interp_indices),
```

Comment:
I'm confused -- why is `__getitem__` on a `ToeplitzLazyVariable` returning an `InterpolatedLazyVariable`?

Reply:
Do the "interpolation" matrices basically just pick out the correct elements?

Reply:
Yeah. My plan was to make this the default `__getitem__` behavior. It's such an annoying method to write for every lazy variable.

Reply:
I think I forgot to drag it out to LazyVariable though.